Adyen/UI/List/CoreDataSource.swift

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+
+/// :nodoc:
+/// A `ListViewController` data source.
+internal protocol ListViewControllerDataSource: UITableViewDataSource {
+    
+    /// :nodoc:
+    /// All sections data.
+    var sections: [ListSection] { get }
+    
+    /// :nodoc:
+    /// Cell reuse idenitifier.
+    var cellReuseIdentifier: String { get }
+    
+    /// :nodoc:
+    /// Reloads all data.
+    func reload(newSections: [ListSection], tableView: UITableView, animated: Bool)
+    
+    /// :nodoc:
+    /// Deletes an item at a certain `IndexPath`.
+    func deleteItem(at indexPath: IndexPath, tableView: UITableView, animated: Bool)
+    
+    /// :nodoc:
+    /// Creates a new `UITableViewCell` at a certain `IndexPath`.
+    func cell(for tableView: UITableView, at indexPath: IndexPath) -> UITableViewCell
+    
+    /// :nodoc:
+    /// Starts a loading animation for a given ListItem.
+    func startLoading(for item: ListItem, _ tableView: UITableView)
+    
+    /// :nodoc:
+    /// Stops all loading animations.
+    func stopLoading(_ tableView: UITableView)
+}
+
+internal final class CoreDataSource: NSObject, ListViewControllerDataSource {
```

Comment:
Maybe the name could be better. `CoreData` usually means something different in iOS dev context :)